src/prog_models/prognostics_model.py

```diff
@@ -1113,21 +1126,78 @@ def calc_error(self, times: List[float], inputs: List[InputContainer], outputs:
         """
         method = kwargs.get('method', 'MSE')
 
-        # Call appropriate error calculation method
-        if method.lower() == 'mse':
-            return calc_error.MSE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'max_e':
-            return calc_error.MAX_E(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'rmse':
-            return calc_error.RMSE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'mae':
-            return calc_error.MAE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'mape':
-            return calc_error.MAPE(self, times, inputs, outputs, **kwargs)
-
-        # If we get here, method is not supported
-        raise ProgModelInputException(f"Error method '{method}' not supported")
+        method_map = {
+            'mse': calc_error.MSE,
+            'max_e': calc_error.MAX_E,
+            'rmse': calc_error.RMSE,
+            'mae': calc_error.MAE,
+            'mape': calc_error.MAPE
+        }
+
+        try:
+            method = method_map[method.lower()]
+        except KeyError:
+            # If we get here, method is not supported
+            raise ProgModelInputException(f"Error method '{method}' not supported")
+        
+        acceptable_types = {dict, tuple, np.ndarray, list, SimResult, LazySimResult}
+        types = {type(times), type(inputs), type(outputs)}
+
+        if not all(t in acceptable_types for t in types):
+            type_error = f"Types passed in must be from the following list: np.ndarray, list, SimResult, or LazySimResult. Current types"
+            type_error += f" at data location {_loc}" if _loc is not None else ""
+            type_error += f": times = {type(times).__name__}, inputs = {type(inputs).__name__}, and outputs = {type(outputs).__name__}"
+            raise TypeError(type_error)
+        if len(times) != len(inputs) or len(inputs) != len(outputs):
+            len_error = "Times, inputs, and outputs must all be the same length. Current lengths"
+            len_error += f" at data location ({_loc})" if _loc is not None else ""
+            len_error += f": times = {len(times)}, inputs = {len(inputs)}, outputs = {len(outputs)}"
+            raise ValueError(len_error)
+        if len(times) < 2:
+            # raise ValueError(f"Must Provide at least 2 data points. Currently only passing in {len(times)}")
+            less_2_error = "Must provide at least 2 data points for times, inputs, and outputs"
+            less_2_error += f" at data location ({_loc})." if _loc is not None else ""
+            raise ValueError(less_2_error)
     
+        x = kwargs.get('x0', self.initialize(inputs[0], outputs[0]))
```

Comment:
Hmm... it is in line 1195 not as a check, but as an object whose property is being accessed, specifically the name of its type.

The check for the initial state is done in line 1194, which is done by accessing kwargs['x0']. What I can do is replace that section with x instead.

I could also just remove line 1161, but I think it is more consistent with lines 1162 and 1163 to leave it as it is and do the change as mentioned above.

Reply:
I would suggest changing 1194 back to what it was originally (just with x0 and not x), and delete the current line 1161. Since it's not necessary here, it's (potentially) an added call to the initialize function that is never used. This line has to be called again within each error method, since kwargs (and not x, dt and stability_tol defined in 1161-1163) are passed to each method. That's just my suggestion, happy to hear from you and @teubert as well!